validator/src/consensus/notifier.rs

```diff
@@ -176,3 +180,106 @@ fn from_hex<T: AsRef<str>>(hex_string: T, id_name: &str) -> Vec<u8> {
         Err(err) => panic!("{} is invalid hex: {:?}", id_name, err),
     }
 }
+
+#[derive(Debug)]
+enum ConsensusNotification {
+    PeerConnected(String),
+    PeerDisconnected(String),
+    PeerMessage((ConsensusPeerMessage, String)),
+    BlockNew(Block),
+    BlockValid(String),
+    BlockInvalid(String),
+    BlockCommit(String),
+    BatchNew(String),
+    BatchInvalid(String),
+}
+
+#[derive(Clone)]
+pub struct BackgroundConsensusNotifier {
+    tx: Arc<Mutex<Sender<ConsensusNotification>>>,
```

Comment:
Sender's don't need to be in a `Mutex`

Reply:
In this context, it does because of the trait bounds on the ConsensusNotifier trait, which are there because of Sync requirements on the BlockPublisher.